File: library_net/src/main/java/com/hjc/library_net/bean/BaseResponse.java
package com.hjc.library_net.bean;

import androidx.annotation.NonNull;

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:46
 * @Description: 通用返回Json封装
 */
public class BaseResponse<T> {
    private String errorCode;
    private String errorMsg;
    private T data;

    public String getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(String errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @NonNull
    @Override
    public String toString() {
        return "BaseResponse{" +
                "errorCode='" + errorCode + '\'' +
                ", errorMsg='" + errorMsg + '\'' +
                ", data=" + data +
                '}';
    }
}

File: library-net/src/main/java/com/hjc/library_net/bean/BaseResponse.kt
package com.hjc.library_net.bean

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:46
 * @Description: 通用Json封装
 */
class BaseResponse<T> {
    var errorCode: Int? = null
    var errorMsg: String? = null
    var data: T? = null

    override fun toString(): String {
        return "BaseResponse{errorCode=$errorCode, errorMsg='$errorMsg', data=$data}"
    }
}
